cf-runagent/cf-runagent.c

```diff
@@ -158,6 +160,41 @@ char REMOTEBUNDLES[CF_MAXVARSIZE] = "";
 
 /*****************************************************************************/
 
+/**
+ * @param is_exit_code whether #remote_exit_status is a exit code directly
+ *                     (#true) or a status from wait() (#false)
+ */
+static inline void UpdateExitCode(int *exit_code, int remote_exit_status, bool one_host, bool is_exit_code)
+{
+    assert(exit_code != NULL);
+
+    if (one_host)
+    {
+        if (is_exit_code)
+        {
+            *exit_code = remote_exit_status;
+            return;
+        }
+        else if (WIFEXITED(remote_exit_status))
+        {
+            *exit_code = WEXITSTATUS(remote_exit_status);
+            return;
+        }
+        else
+        {
+            *exit_code = CF_RA_EXIT_CODE_OTHER_ERR;
+            return;
+        }
```

Comment:
Nitpick, but it's easier for me to see that this part is guaranteed to return if you skip the elses:

```suggestion
        if (WIFEXITED(remote_exit_status))
        {
            *exit_code = WEXITSTATUS(remote_exit_status);
            return;
        }
        *exit_code = CF_RA_EXIT_CODE_OTHER_ERR;
        return;
```